lib/setup/profile.js

```diff
@@ -66,7 +69,104 @@ var appendScript = function(name, options) {
 };
 
 /**
- * Update `~/.bash_profile` and `.zshrc` shell profile files.
+ * Chooses the correct shell init script to insert avn initialization string to,
+ * based on the following criteria:
+ * 1. Append to the one that's present, if the other doesn't exist,
+ * 2. If both exist, append to the one that is sourced by the other, or
+ * 3. If both exist, but neither sources the other,
+ *    append to .bashrc if platform is 'linux',
+ *    or append to .bash_profile if platform is 'darwin' (i.e. macOS)
+ *
+ * @private
+ * @function setup.profile.~shellInitScriptChooser
+ * @return {Promise}
+ */
+var shellInitScriptChooser = function() {
+  var bashrcFile = path.join(process.env.HOME, bashrcFileName);
+  var bashProfileFile = path.join(process.env.HOME, bashProfileFileName);
+  var bashProfileNotExist;
+  var bashrcNotExist;
+
+  // Notice: will change fs.stat to fs.access in the future
+  // Reason: NodeJS 0.10 doesn't have fs.access (added in NodeJS 0.11.15)
+
+  // Check if .bash_profile exists
+  return fs.stat(bashProfileFile)
+  .catch(function(err) {
+    bashProfileNotExist = err;
+  })
+
+  // Check if .bashrc exists
+  .then(function() {
+    return fs.stat(bashrcFile)
+    .catch(function(err) {
+      bashrcNotExist = err;
+    });
+  })
+
+  .then(function() {
+    if (!bashProfileNotExist && !bashrcNotExist) {
+      return fs.readFile(bashProfileFile, 'utf8')
+      .then(function(data) {
+        // RegExp to detect "~/.bashrc",
+        // "$HOME/.bashrc", or "/home/user/.bashrc"
+        var bashrcRe = new RegExp(' ?source +(\\\\\\n)? *(~\/.bashrc|' +
```

Comment:
- `.` is an alias of `source`
- All space checks should probably use `\s`
- Whitespace checks that can span a line should use the multiline workaround of `[\s\S]`
- Checking for the full path seems unnecessary (`~` vs `$HOME` vs `/platform/specific/prefix`) and error prone… for instance, I could write `${HOME}` or `${HOME:/home}`.